static/js/jquery.pjax.modified.js

```diff
@@ -130,6 +126,15 @@
         if (event.isDefaultPrevented())
             return
 
+        // Close a modal window if the link with provided URL is opened from the modal window
+        // in order to display the loaded page content in background under the modal
+        if (link.href.length && link.href !== '#') {
```

Comment:
I still prefer a solution similar to the one I mentioned in https://github.com/humhub/humhub/issues/2950#issuecomment-770648113 since:

- This solution does only work on click events but pjax can also be triggered directly by javascript e.g. pjax.redirect() or pjax.reload()
- There are potentially multiple modals open at once
- The jquery.pjax.modified is a patched version of the original I'd rather prevent custom code here if possible and also don't mix in some modal specific code

Reply:
Ok, please check commit https://github.com/humhub/humhub/pull/4812/commits/091704f1b637fee09fb4ed16554103ef93435df4

Reply:
But I see many tests are failed here https://github.com/humhub/humhub/runs/1807645931?check_suite_focus=true, I have to fix them.

Reply:
The failed tests work without errors on my local server:

<img width="1428" alt="test_activity" src="https://user-images.githubusercontent.com/6995524/106561712-3bc1d780-653a-11eb-8365-1667cbed6fa7.png">
<img width="1484" alt="test_space" src="https://user-images.githubusercontent.com/6995524/106561730-43817c00-653a-11eb-9904-d4c684e905c6.png">
<img width="1274" alt="test_stream" src="https://user-images.githubusercontent.com/6995524/106561743-47ad9980-653a-11eb-8830-91990304cada.png">
<img width="1373" alt="test_user" src="https://user-images.githubusercontent.com/6995524/106561746-49775d00-653a-11eb-820b-86d9e77411e2.png">

Reply:
Maybe the acceptance tests run a bit faster in the git test environment and this maybe leads to race condition. Are those errors related to modal?